src/common.rs

```diff
@@ -1291,28 +1326,39 @@ pub fn print_transaction_status(
             }
         }
         near_primitives::views::FinalExecutionStatus::SuccessValue(bytes_result) => {
-            eprintln!("--- Result -------------------------");
-            if bytes_result.is_empty() {
-                eprintln!("Empty result");
+            let result_output = if bytes_result.is_empty() {
+                "Empty result".to_string()
             } else if let Ok(json_result) =
                 serde_json::from_slice::<serde_json::Value>(bytes_result)
             {
-                println!("{}", serde_json::to_string_pretty(&json_result)?);
+                serde_json::to_string_pretty(&json_result)?
             } else if let Ok(string_result) = String::from_utf8(bytes_result.clone()) {
-                println!("{string_result}");
+                string_result
             } else {
-                eprintln!("The returned value is not printable (binary data)");
-            }
-            eprintln!("------------------------------------\n");
-            print_value_successful_transaction(transaction_info.clone());
+                "The returned value is not printable (binary data)".to_string()
+            };
+            if let crate::Verbosity::Quiet = verbosity {
+                std::io::stdout().write_all(bytes_result)?;
+                return Ok(());
+            };
```

Comment:
Move this block before `let result_output`, so the program does not perform useless work.

Reply:
Done